spec/rollout_spec.rb

```diff
@@ -89,6 +89,23 @@
     end
   end
 
+  ##
+  # CONTRACT: assess feature status for the user
+  #
+  # assumming client uses it like this rollout.active?('feature_name', user)
+  #
+  # accepts feature name and some kind of identifiable object
+  #
+  # returns boolean: true or false
+  # * true when feature is active for the user
+  # * false when feature isn't active for the user
+  #
+  # According to current implementation
+  #   feature name can be a string or symbol
+  #   user can be object with ID attribute, number or string identifier
+  #
+  # But I think actual client is more specific
```

Comment:
Interesting! What does this mean? Maybe say more details here before the knowledge disappears from your mind. :)

Reply:
I had some doubts between identifiable and "always a user", just trying to say that client most likely interested in a specific identifiable, i.e. user, integer or string and not all at once. I'll update contract accordingly :)

Reply:
Yes. The phrase "But I think actual client is more specific" confused me. I believe that I understand your goal: "I don't need a user here, but rather anything with an :id property. Even so, I believe that all clients currently always pass me a user here."

I really like your impulse to depend on the fewest details possible.

Reply:
I wonder if all we need here is ID or UID? :)

Current implementation accepts number, string or object which responds to ID property, where ID property can also be number or string. So, it seems like this object.id is a kind of an extra convenience supplier provides to its clients, perhaps assuming that clients often have to deal with such kind of objects.

But, from the Client perspective, is it sensible to send entire object when only one property is needed? Also, if we stick with "always a user", there can be scenarios when Client will need to get user somewhere, for example `rollout.active?('some_feature_name', User.find(id))`.

I'll appreciate any thoughts / advices about this :)

Thank you in advance!